crates/reporting/src/report.rs

```diff
@@ -1652,6 +1652,35 @@ pub fn to_file_problem_report<'b>(
                 severity: Severity::Fatal,
             }
         }
+        io::ErrorKind::Unsupported => {
+            let doc = match filename.extension() {
+                Some(ext) => {
+                    alloc.concat(vec![
+                        alloc.reflow(
+                            r"I am expecting a roc application file with either `.roc` or no extension. Instead I found a file with extension `.",
+                        ),
+                        alloc.as_string(ext.to_string_lossy()),
+                        alloc.as_string("`"),
+                    ])
+                }
+                None => {
+                    alloc.stack(vec![
+                        alloc.concat(vec![
+                            alloc.reflow(r"I am expecting a roc application file with either `.roc` or no extension and a shebang directive. "),
+                            alloc.reflow(r"Instead I found a file without an extension and without a shebang"),
+                        ]),
+                        alloc.reflow(r"Consider starting the file with `#!` and the path to your roc binary"),
+                    ])
+                }
+            };
+
+            Report {
+                filename,
+                doc,
+                title: "EXPECTED ROC FILE".to_string(),
```

Comment:
I'd like to go with "NOT A ROC FILE" here. We prefer to keep the wording as simple as possible for non-native English speakers. It's also shorter :)